app/models/custom/officing/residence.rb

```diff
@@ -0,0 +1,59 @@
+require_dependency Rails.root.join("app", "models", "officing", "residence").to_s
+
+class Officing::Residence
+  def save
+    return false unless valid?
+
+    if user_exists?
+      self.user = find_user_by_document
+      user.update!(verified_at: Time.current)
+    else
+      user_params = {
+        document_number:       document_number,
+        document_type:         document_type,
+        geozone:               geozone,
+        date_of_birth:         birthday&.in_time_zone&.to_datetime,
+        gender:                gender,
+        residence_verified_at: Time.current,
+        verified_at:           Time.current,
+        erased_at:             Time.current,
+        password:              random_password,
+        terms_of_service:      "1",
+        email:                 nil
+      }
+      self.user = User.create!(user_params)
+    end
+  end
+
+  def allowed_age?
+    return false if birthday.blank?
+
+    Age.in_years(birthday) >= User.minimum_required_age
```

Comment:
A question about something you mention in the commit message:

> it is not verified with the web service so users could lie about its age.

Correct if I'm wrong about this.

Let's say I'm 14 years old but lie and say I'm 24 years old. When that happens, the application sends a request to the remote census API asking for a user with my document number and born 24 years ago, and won't find it, so lying about my age won't get me anywhere.

Is that right, or there's a scenario where users can actually find a loophole in the verification process?

Reply:
The Santiago census web service only needs the document number and type to validate the user residency, it does not ask the users `date_of_birth`, nor the `postal_code` and does not return those data on the web service response, so we do not have a way to check if the user has the minimum allowed age to participate.

This PR implementation asks the user for the `date_of_birth` during the verification process to jump over the minimum allowed age control. We check the entered date is above the minimum allowed age. Still, it could not be the truth as it is not compared with any reliable source of information.